contracts/staking/StakeManager.sol

```diff
@@ -165,6 +176,40 @@ contract StakeManager is Validator, IStakeManager, RootChainable, Lockable {
     emit ReStaked(validatorId, validators[validatorId].amount, totalStaked);
   }
 
+  function claimRewards(uint256 validatorId, uint256 accountBalance, uint256 index, bytes memory proof) public /*onlyStaker(validatorId) */ {
+    // accountState = keccak256(abi.encodePacked(validatorId, accountBalance))
+    require(keccak256(abi.encodePacked(validatorId, accountBalance)).checkMembership(index, accountStateRoot, proof));
+    uint256 _reward = accountBalance.sub(validators[validatorId].totalReward);
+    address _contract = validators[validatorId].contractAddress;
+    if (_contract == address(0x0)) {
+      validators[validatorId].reward = validators[validatorId].reward.add(_reward);
+    }
+    else {
+      // TODO: delegator bond/share rate if return needs to be updated periodically
+      // otherwise validator can delay and get all the delegators reward
+      ValidatorContract(_contract).updateRewards(_reward, currentEpoch, validators[validatorId].amount);
+    }
+    totalRewardsLiquidated += _reward;
+    require(totalRewardsLiquidated <= totalRewards, "Oops this shouldn't have happened");// pos 2/3+1 is colluded
```

Comment:
Error message is inconclusive.

Reply:
hoping this never comes, should I add `pos 2/3+1 has colluded` to the require msg?

Reply:
Lol, it's still incorrect. Nevermind.